rbd/locks_test.go

```diff
@@ -0,0 +1,142 @@
+//go:build !nautilus && ceph_preview
+// +build !nautilus,ceph_preview
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLocking(t *testing.T) {
+	conn := radosConnect(t)
+	require.NotNil(t, conn)
+	defer conn.Shutdown()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	require.NoError(t, err)
+	defer conn.DeletePool(poolname)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx.Destroy()
+
+	// Connect another cluster
+	conn1 := radosConnect(t)
+	require.NotNil(t, conn1)
+	defer conn1.Shutdown()
+
+	ioctx1, err := conn1.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx1.Destroy()
+
+	options := NewRbdImageOptions()
+	defer options.Destroy()
+	assert.NoError(t, options.SetUint64(ImageOptionOrder, uint64(testImageOrder)))
+
+	name := GetUUID()
+	err = CreateImage(ioctx, name, testImageSize, options)
+	assert.NoError(t, err)
+
+	t.Run("acquireLock", func(t *testing.T) {
+		img, err := OpenImage(ioctx, name, NoSnapshot)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, img.Close())
+		}()
+
+		// Shared lock mode is not supported
```

Comment:
Not at all supported? Can you elaborate?

Reply:
Good find, also a good link to  add to your comment in the code.

Reply:
Updated the commen

Reply:
> This is what I found in the ceph codebase, ref: [lock_acquire](https://github.com/ceph/ceph/blob/afb6b17aa77f7a85bf6d28b0174fa3c8609efc0d/src/librbd/internal.cc#L928-L929)

Interesting one. They define and say "no, you can't use it".